parquet/src/main/java/org/apache/iceberg/parquet/MessageTypeToType.java

```diff
@@ -96,11 +96,13 @@ public Type struct(GroupType struct, List<Type> fieldTypes) {
 
   @Override
   public Type list(GroupType array, Type elementType) {
-    GroupType repeated = array.getType(0).asGroupType();
-    org.apache.parquet.schema.Type element = repeated.getType(0);
+    org.apache.parquet.schema.Type repeated = array.getType(0);
+    org.apache.parquet.schema.Type repeatedElement = array.getFields().get(0);
+    boolean isElementType = ParquetSchemaUtil.isListElementType(repeatedElement, array.getName());
+    org.apache.parquet.schema.Type element = isElementType ? repeated : repeated.asGroupType().getType(0);
 
     Preconditions.checkArgument(
-        !element.isRepetition(Repetition.REPEATED),
+        isElementType || !element.isRepetition(Repetition.REPEATED),
         "Elements cannot have repetition REPEATED: %s", element);
```

Comment:
Should we change this message as well? Since now elements can be repeated but only in backwards compatibility situations?

Reply:
I don't know that this is useful anymore. I think it was actually checking to make sure we didn't try to read 2-level lists.